src/sagas/ses.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { MMKV } from 'react-native-mmkv';
+import RNRestart from 'react-native-restart';
+import {
+  call,
+  fork,
+  take,
+  all,
+} from 'redux-saga/effects';
+import { checkForFeatureFlag } from './helpers';
+import { SES_FEATURE_TOGGLE, SHOULD_ENABLE_SES_STORAGE_KEY } from '../constants';
+
+const storage = new MMKV();
+
+function verifySesEnabled() {
+  return Object.isFrozen([].prototype);
+}
+
+function disableSes(restart = true) {
+  console.log('disabling SES!!');
+  storage.set(SHOULD_ENABLE_SES_STORAGE_KEY, false);
+
+  if (restart) {
+    RNRestart.restart();
+  }
+}
+
+function enableSes(restart = true) {
+  storage.set(SHOULD_ENABLE_SES_STORAGE_KEY, true);
+
+  if (restart) {
+    RNRestart.restart();
+  }
+}
+
+/**
+ * `init` will run on the app initialization.
+ *
+ */
+function* init() {
+  const storageEnabled = storage.getBoolean('should-enable-ses');
+  const unleashEnabled = yield call(checkForFeatureFlag, SES_FEATURE_TOGGLE);
+  const sesEnabled = verifySesEnabled();
+
+  if (unleashEnabled && storageEnabled && !sesEnabled) {
+    // This is an issue, the environment is not secure, we should issue a fatal
+    // error.
```

Comment:
question(non-blocking): Why don't we throw here at the current code?

Reply:
Sorry, I removed the error report from the code and forgot to re-add it

Fixed in 3c7a1286436cbea8a5e1ed5fde1ced0c0a6740c9